lib/rspec/core/example_group.rb

```diff
@@ -424,11 +424,15 @@ def self.set_it_up(description, args, registration_collection, &example_group_bl
           superclass.method(:next_runnable_index_for),
           description, *args, &example_group_block
         )
+
+        config = RSpec.configuration
+        config.apply_derived_metadata_to(@metadata)
```

Comment:
Is there a way to deduplicate this somehow? Could we roll it into `RSpec.configuration` with a bit of memoization or is there a chicken and egg problem.

Reply:
I don't see how.  We've already rolled the logic into `RSpec.configuration` (in the form of `RSpec.configuration.apply_derived_metadata_to(@metadata)`), but we need to call it from somewhere after `@metadata` has been assigned.

You're welcome to give it a try in a follow up PR :).